app/services/proforma_service/convert_exercise_to_task.rb

```diff
@@ -77,62 +77,85 @@ def uuid
     end
 
     def model_solutions
-      @exercise.files.filter {|file| file.role == 'reference_implementation' }.map do |file|
+      @exercise.files.filter {|file| file.role == 'reference_implementation' }.group_by {|file| file.xml_id_path&.split('/')&.first || "co-ms-#{file.id}" }.map do |xml_id, files|
```

Comment:
Any chance to split this line into dedicated part to improve readability? I got what it's doing, but it seems pretty complex.

<hr>

And why is the `xml_id_from_file(file)` method not used?

Reply:
> And why is the `xml_id_from_file(file)` method not used?

it somehow escaped my refactoring :eyes: